controllers/usersignup/space.go

```diff
@@ -0,0 +1,45 @@
+package usersignup
+
+import (
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func newSpace(userSignup *toolchainv1alpha1.UserSignup, targetCluster targetCluster, compliantUserName, tier string) *toolchainv1alpha1.Space {
+	labels := map[string]string{
+		toolchainv1alpha1.SpaceCreatorLabelKey: compliantUserName,
```

Comment:
Right, that makes sense since the UserSignup is creating it, not the MUR.

Reply:
Yes, please use the UserSignup name. It will be slightly weird because the SpaceBindings will use the MUR name, but it will be safer for other cases.

If we used the compliant username, then it could get complicated in situations like:

1. I'm provisioned with compliant username mjobanek
2. By default, I'm allowed to create only one Space 
3. I create a Space
4. I share it with another user and grant admin permissions to the user
5. I let the admins delete my account (for example as part of GDPR deletion)
6. The Space can stay present because it has another admin
7. The Space still contains the `creator` label, but it refers to `mjobanek` MUR that doesn't exist anymore
8. There is a new  user provisioned whose compliant username would match the same name as I used before: `mjobanek`

I know that it's a very small corner case, but referring to a `UserSignup` will give us the ability to always find the correct `UserSignup` that created it. Also, there is no place for any conflicts of the names because the name of the `UserSignup` should be always unique and should always be generated from only one particular SSO user.
cc @alexeykazakov